dcpy/connectors/edm/publishing.py

```diff
@@ -142,39 +168,153 @@ def legacy_upload(
 def publish(
     product: str,
     build: str,
-    publishing_version: str,  ## TODO - ideally this is optional, and we have mechanisms for determining programmatically
-    acl: str,
     *,
+    acl: str,
+    publishing_version: Optional[str] = None,
     keep_draft: bool = True,
     max_files: int = 30,
 ):
     """Publishes a specific draft build of a data product
     By default, keeps draft output folder"""
+    if publishing_version is None:
+        publishing_version = get_draft_version(product, build)
     source = f"{product}/draft/{build}/"
     target = f"{product}/publish/{publishing_version}/"
     s3.copy_folder(BUCKET, source, target, acl, max_files=max_files)
     if not keep_draft:
         s3.delete(BUCKET, source)
 
 
-def read_csv(dataset, version, filepath, **kwargs):
-    """Reads csv into pandas dataframe from edm-publishing
-    'version' currently must be a path from root dataset folder in edm-publishing to output folder
+def _read_data_helper(path: str, filereader, **kwargs):
+    with s3.get_file_as_stream(BUCKET, path) as stream:
+        data = filereader(stream, **kwargs)
+    return data
+
+
+def read_csv(product: str, version: str, filepath: str, **kwargs):
```

Comment:
would be nice to add expected kwargs to the docstring

Reply:
See latest commit for updated docstrings, type signatures